core/javax.cache/src/main/java/org/wso2/carbon/caching/impl/CacheCleanupTask.java

```diff
@@ -58,15 +58,19 @@ public synchronized void run() {
             cc.setTenantId(MultitenantConstants.SUPER_TENANT_ID);
             cc.setTenantDomain(MultitenantConstants.SUPER_TENANT_DOMAIN_NAME);
             for (CacheImpl cache : caches) {
-                cache.runCacheExpiry();
-                if (log.isDebugEnabled()) {
-                    log.debug("Cache expiry completed for cache " + cache.getName());
+                try {
+                    cache.runCacheExpiry();
+                    if (log.isDebugEnabled()) {
+                        log.debug("Cache expiry completed for cache " + cache.getName());
+                    }
+                } catch (IllegalStateException e) {
```

Comment:
We can omit this catch block and just catch the throwable?

Reply:
The log level of each of the catch clauses is different. Therefore we cannot handle `IllegalStateException` from the catch block of `Throwable`